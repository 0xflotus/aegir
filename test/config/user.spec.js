/* eslint-env jest */
'use strict'

describe('config - user', () => {
  let config
  beforeAll(() => {
    jest.mock('../../src/utils', () => ({
      getPkg () {
        return Promise.resolve({
          name: 'example'
        })
      },
      getUserConfig () {
        return {
          webpack: {
            devtool: 'eval'
          },
          entry: 'src/main.js'
        }
      },
      getLibraryName () {
        return 'Example'
      },
      getPathToDist () {
        return 'dist'
      },
      getPathToNodeModules () {
        return 'aegir/node_modules'
      }
    }))
    config = require('../../src/config/user')()
  })

  afterAll(() => {
    jest.resetAllMocks()
  })

  it('custom config', () => {
    expect(config).toMatchSnapshot()

    expect(config).toHaveProperty('webpack', {
      devtool: 'eval'
    })
    expect(config).toHaveProperty('entry', 'src/main.js')
  })
})
