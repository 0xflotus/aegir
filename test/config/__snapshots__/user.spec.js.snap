// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config - user custom config 1`] = `
Object {
  "entry": "src/main.js",
  "karma": Object {},
  "webpack": Object {
    "devtool": "eval",
  },
}
`;
